webclient_upload/src/main/java/com/baeldung/reactive/webclient/upload/controller/UploadController.java

```diff
@@ -0,0 +1,43 @@
+package com.baeldung.reactive.webclient.upload.controller;
+
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import reactor.core.publisher.Mono;
+
+
+@RestController
+public class UploadController {
+    final ReactiveUploadService uploadService;
+
+    public UploadController(ReactiveUploadService uploadService) {
+        this.uploadService = uploadService;
+    }
+
+    @PostMapping(path = "/upload")
+    @ResponseBody
+    public Mono<HttpStatus> uploadPdf(@RequestParam("file") final MultipartFile multipartFile) {
+        return uploadService.uploadPdf(multipartFile.getResource());
+    }
+
+    @PostMapping(path = "/upload/multipart")
+    @ResponseBody
+    public Mono<HttpStatus> uploadMultipart(@RequestParam("file") final MultipartFile multipartFile) {
+        return uploadService.uploadMultipart(multipartFile);
+    }
+
+
+    /**
+     * Fake upload endpoint returning "OK" HttpStatus
+     * @return "OK" HttpStatus
+     */
+    @PostMapping(path = "/external/upload")
+    @ResponseBody
+    public HttpStatus externalUpload() {
+        return HttpStatus.OK;
+    }
+}
```

Comment:
Not too sure if this is enough to demonstrate that it works. 
If we properly want to create an upload location we could use a localstack container and an s3 bucket or something, but it might be an overkill.